sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/nullExpressions.scala

```diff
@@ -91,29 +91,36 @@ case class Coalesce(children: Seq[Expression]) extends Expression {
     val codes = ctx.splitExpressionsWithCurrentInputs(
       expressions = evals,
       funcName = "coalesce",
-      makeSplitFunction = func =>
-        s"""
-           |do {
-           |  $func
-           |} while (false);
-         """.stripMargin,
-      foldFunctions = _.map { funcCall =>
-        s"""
-           |$funcCall;
-           |if (!${ev.isNull}) {
-           |  continue;
-           |}
-         """.stripMargin
-      }.mkString)
-
+      returnType = ctx.javaType(dataType),
+      makeSplitFunction = {
+        func =>
```

Comment:
ditto, keep previous code style.

Reply:
Ah, got it about `makeSplitFunction = func =>`. I thought about multline style. Sorry for my misunderstanding.